scripts/exglobal_diag.sh

```diff
@@ -170,113 +170,126 @@ EOFdiag
                   else
-                     $CATEXEC -o diag_${type}_${string}.${CDATE}${DIAG_SUFFIX} ${prefix}${type}_${loop}*
+                     "${CATEXEC}" -o "diag_${type}_${string}.${CDATE}${DIAG_SUFFIX}" "${prefix}${type}_${loop}*"
                   fi
                fi
-               echo "diag_${type}_${string}.${CDATE}*" >> ${diaglist[n]}
-               numfile[n]=$(expr ${numfile[n]} + 1)
-            elif [ $count -eq 1 ]; then
-                cat ${prefix}${type}_${loop}* > diag_${type}_${string}.${CDATE}${DIAG_SUFFIX}
-                if [ $DIAG_COMPRESS = "YES" ]; then
-                   $COMPRESS diag_${type}_${string}.${CDATE}${DIAG_SUFFIX}
+               echo "diag_${type}_${string}.${CDATE}*" >> "${diaglist[n]}"
+               numfile[n]=$(expr "${numfile[n]}" + 1)
+            elif [[ "${count}" -eq 1 ]]; then
+                cat "${prefix}${type}_${loop}*" > "diag_${type}_${string}.${CDATE}${DIAG_SUFFIX}"
+                if [[ "${DIAG_COMPRESS}" = "YES" ]]; then
+                   "${COMPRESS}" "diag_${type}_${string}.${CDATE}${DIAG_SUFFIX}"
                 fi
-                echo "diag_${type}_${string}.${CDATE}*" >> ${diaglist[n]}
-                numfile[n]=$(expr ${numfile[n]} + 1)
+                echo "diag_${type}_${string}.${CDATE}*" >> "${diaglist[n]}" 
+                numfile[n]=$(expr "${numfile[n]}" + 1)
             fi
          done
       done
-      echo $(date) END loop $string >&2
+      echo "$(date)" END loop "${string}" >&2 || true
    done
 
    # We should already be in $DATA, but extra cd to be sure.
-   cd $DATA
+   cd "${DATA}" || exit
 
    # If requested, compress diagnostic files
-   if [ $DIAG_COMPRESS = "YES" -a $USE_CFP = "NO" ]; then
-      echo $(date) START $COMPRESS diagnostic files >&2
-      for file in $(ls diag_*${CDATE}${DIAG_SUFFIX}); do
-         $COMPRESS $file
+   if [[ "${DIAG_COMPRESS}" = "YES" && "${USE_CFP}" = "NO" ]]; then
+      echo "$(date)" START "${COMPRESS}" diagnostic files >&2 || true
+      # shellcheck disable=SC2045 
+      for file in $(ls "diag_*${CDATE}${DIAG_SUFFIX}"); do
+         "${COMPRESS}" "${file}"
       done
-      echo $(date) END $COMPRESS diagnostic files >&2
+      echo "$(date)" END "${COMPRESS}" diagnostic files >&2 || true
    fi
 
-   if [ $USE_CFP = "YES" ] ; then
-      chmod 755 $DATA/mp_diag.sh
-      ncmd=$(cat $DATA/mp_diag.sh | wc -l)
-      if [ $ncmd -gt 0 ]; then
+   if [[ "${USE_CFP}" = "YES" ]] ; then
+      chmod 755 "${DATA}/mp_diag.sh"
+      # shellcheck disable=SC2002,SC2312 
+      ncmd=$(cat "${DATA}/mp_diag.sh" | wc -l)
+      # shellcheck disable=SC2034
+      if [[ "${ncmd}" -gt 0 ]]; then
          ncmd_max=$((ncmd < npe_node_max ? ncmd : npe_node_max))
+         # shellcheck disable=SC2086,SC2250 
          APRUNCFP_DIAG=$(eval echo $APRUNCFP)
-         $APRUNCFP_DIAG $DATA/mp_diag.sh
+         "${APRUNCFP_DIAG}" "${DATA}/mp_diag.sh"
```

Comment:
```suggestion
         "${APRUNCFP}" "${DATA}/mp_diag.sh"
```

Reply:
Actually, that won't work. Use
```suggestion
	     APRUNCFP_DIAG=$(echo ${APRUNCFP} | ncmd="${ncmd}" envsubst)
	     ${APRUNCFP_DIAG} "${DATA}/mp_diag.sh"
```